src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryUntypedObject.cs

```diff
@@ -12,6 +13,11 @@ namespace Microsoft.PowerFx.Functions
 {
     internal static partial class Library
     {
+        private static bool IsValidDateTimeUO(string s)
+        {
+            return Regex.IsMatch(s, @"^[0-9]{4,4}(-[0-1][0-9](-[0-3][0-9](T[0-9]{2,2}(:[0-9]{2,2}(:[0-9]{2,2}(\.[0-9]{3,3}Z)?)?)?)?)?)?$");
```

Comment:
>Regex [](http://example.com/codeflow?start=19&length=5)

This regex accepts just a year ("2022"), which shouldn't be

Reply:
Same problem with hour only in the time part ("2022-08-03T10")

Reply:
Fixed